### PR TITLE
Add fromUrl and fromUrlStatus methods

### PR DESCRIPTION
**Pull request by bautrukevich**

### CODE REVIEW

**Comment by bautrukevich** [comment]
[UploadClient: wrappers from url paths of Upload API](https://trello.com/c/MhDdymTo/252-uploadclient-wrappers-from-url-paths-of-upload-api)

File: src/api/fromUrl.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * @param {string} sourceUrl – Source file URL, which should be a public HTTP or HTTPS link.
+ * @param {Settings} settings
+ * @return {Promise<FromUrlResponse>}
+ */
+export default function fromUrl(sourceUrl: string, settings: Settings = {}): Promise<FromUrlResponse> {
+  const options: RequestOptions = prepareOptions({
+    path: '/from_url/',
+    query: {
+      pub_key: settings.publicKey || '',
+      source_url: sourceUrl,
```
Comment (Zmoki) [inline src/api/fromUrl.ts]: `/from_url/` has more parameters, see https://api-refs.uploadcare.com/upload-api.html#operation/multipartFileUploadComplete

File: src/api/fromUrlStatus.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import request, {prepareOptions} from './request'
+import {RequestOptions} from './request'
+import {InfoResponse} from './info'
+import {Settings} from '../types'
+
+export type FromUrlStatusResponse = InfoResponse
```
Comment (Zmoki) [inline src/api/fromUrlStatus.ts]: nope, response difference, InfoResponse is info about the file, FromUrlStatusResponse is info about the upload progress, see https://api-refs.uploadcare.com/upload-api.html#operation/fromURLUploadStatus

File: test/api/fromUrl.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import fromUrl from '../../src/api/fromUrl'
+import * as factory from '../_fixtureFactory'
+
+describe('API - from url', () => {
+  it('should return token for file', async() => {
+    const data = await fromUrl(factory.imageUrl('valid'), {publicKey: factory.publicKey('demo')})
+
+    expect(data.uuid).toBeTruthy()
```
Comment (Zmoki) [inline test/api/fromUrl.spec.js]: `/from_url/` always return only `type` attribute, if `type` is `token`, response contain token, if - `file_info`, response also contain info about file, see https://uploadcare.com/docs/api_reference/upload/from_url/#return-values

So, you shoul check `type=token` and token attribute in response

File: test/api/fromUrlStatus.spec.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import fromUrlStatus from '../../src/api/fromUrlStatus'
+import * as factory from '../_fixtureFactory'
+
+describe('API - from url status', () => {
+  it('should return info about file uploaded from url', async() => {
+    const data = await fromUrlStatus(factory.token('valid'))
+
+    expect(data.uuid).toBeTruthy()
```
Comment (Zmoki) [inline test/api/fromUrlStatus.spec.js]: from url status returns response with upload info, see review comment in fromUrlStatus

File: src/api/fromUrl.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+export default function fromUrl(
+  {sourceUrl, checkForUrlDuplicates, saveUrlForRecurrentUploads}: UrlData, settings: Settings = {}
+): Promise<FromUrlResponse> {
+  const options: RequestOptions = prepareOptions({
+    method: 'POST',
+    path: '/from_url/',
+    query: {
+      pub_key: settings.publicKey || '',
+      source_url: sourceUrl,
+      store: settings.doNotStore ? settings.doNotStore : true,
```
Comment (Zmoki) [inline src/api/fromUrl.ts]:
```suggestion
      store: settings.doNotStore ? '' : 'auto',
```

**Pull request merged**